core/src/main/java/org/apache/iceberg/MetricsModes.java

```diff
@@ -144,4 +146,22 @@ public String toString() {
       return "full";
     }
   }
+
+  private abstract static class ProxySerializableMetricsMode implements MetricsMode {
```

Comment:
We use a similar approach in `PrimitiveType`. If we serialize `MetricsMode` directly, it will be deserialized back on executors and we will have a problem that `MetricsMode.None.get()` is never equal to our deserialized value even though it was `none` before as use `==` for equality checks.

Reply:
Can you add this as a comment in the code so we don't forget this?

Reply:
Done